cmd/cleanup-controller/handlers/cleanup/handlers.go

```diff
@@ -108,25 +115,44 @@ func (h *handlers) lookupPolicy(namespace, name string) (kyvernov2alpha1.Cleanup
 	}
 }
 
-func (h *handlers) executePolicy(ctx context.Context, logger logr.Logger, policy kyvernov2alpha1.CleanupPolicyInterface, cfg config.Configuration) error {
+func (h *handlers) executePolicy(
+	ctx context.Context,
+	logger logr.Logger,
+	policy kyvernov2alpha1.CleanupPolicyInterface,
+	cfg config.Configuration,
+) error {
 	spec := policy.GetSpec()
 	kinds := sets.New(spec.MatchResources.GetKinds()...)
 	debug := logger.V(4)
 	var errs []error
-	enginectx := enginecontext.NewContext(h.jp)
 
-	if spec.Context != nil {
-		for _, entry := range spec.Context {
-			if entry.APICall != nil {
-				if err := engineapi.LoadAPIData(ctx, h.jp, logger, entry, enginectx, h.client); err != nil {
-					return err
-				}
-			} else if entry.Variable != nil {
-				if err := engineapi.LoadVariable(logger, h.jp, entry, enginectx); err != nil {
-					return err
-				}
-			}
-		}
+	enableDeferredLoading := toggle.FromContext(ctx).EnableDeferredLoading()
+	enginectx := enginecontext.NewContext(h.jp, enableDeferredLoading)
+	rclient, err := registryclient.New()
```

Comment:
initialisation is not complete, we should take care of credential helpers, secrets, etc...

Reply:
I will remove it for now, as it seems like image data context entries are not supported in the cleanup controller.

Created this issue to track: https://github.com/kyverno/kyverno/issues/7647